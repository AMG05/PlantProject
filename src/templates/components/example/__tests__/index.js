import { h } from 'preact';
import Example, { Ghost, Empty, Error } from '../';
import render from 'preact-render-to-json';

test('Example > Ghost matches snapshot', () => {
    const tree = render(<Ghost />);
    expect(tree).toMatchSnapshot();
});

test('Example > Empty matches snapshot', () => {
    const tree = render(<Empty />);
    expect(tree).toMatchSnapshot();
});

test('Example > Error matches snapshot', () => {
    const tree = render(<Error />);
    expect(tree).toMatchSnapshot();
});

test('Example > Example matches snapshot', () => {
    const tree = render(<Example href={'#'}
	title={'Quick brown fox'}
	summary={'Jumps over the lazy dog'}
                        >Woof</Example>);
    expect(tree).toMatchSnapshot();
});

