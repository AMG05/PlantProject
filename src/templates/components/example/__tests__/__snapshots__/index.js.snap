// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example > Empty matches snapshot 1`] = `
Object {
  "__": null,
  "__b": 0,
  "__c": null,
  "__d": undefined,
  "__e": null,
  "__k": null,
  "constructor": undefined,
  "key": null,
  "props": Object {},
  "ref": null,
  "type": [Function],
}
`;

exports[`Example > Error matches snapshot 1`] = `
Object {
  "__": null,
  "__b": 0,
  "__c": null,
  "__d": undefined,
  "__e": null,
  "__k": null,
  "constructor": undefined,
  "key": null,
  "props": Object {},
  "ref": null,
  "type": [Function],
}
`;

exports[`Example > Example matches snapshot 1`] = `
Object {
  "__": null,
  "__b": 0,
  "__c": null,
  "__d": undefined,
  "__e": null,
  "__k": null,
  "constructor": undefined,
  "key": undefined,
  "props": Object {
    "children": "Woof",
    "href": "#",
    "summary": "Jumps over the lazy dog",
    "title": "Quick brown fox",
  },
  "ref": undefined,
  "type": [Function],
}
`;

exports[`Example > Ghost matches snapshot 1`] = `
Object {
  "__": null,
  "__b": 0,
  "__c": null,
  "__d": undefined,
  "__e": null,
  "__k": null,
  "constructor": undefined,
  "key": null,
  "props": Object {},
  "ref": null,
  "type": [Function],
}
`;
